### PR TITLE
Master 1158119

### PR DESCRIPTION
**Pull request by schubi2**
## Problem

Keyboard has to be set in the proposal if AutoYaST has not already set it.

* Fast-track: https://trello.com/c/sL8vbVZI/1465-https-bugzillasusecom-showbugcgiid1158119

* This one is blocking Staging:Y from being accepted.

* https://openqa.suse.de/tests/3646052#step/installation/4

## Solution

Avoid to crash when no language and keyboard is defined.

### CODE REVIEW

File: keyboard/src/modules/Keyboard.rb [hunk 1]
```diff
@@ -316,6 +326,7 @@ def current_kbd
     publish :function => :Modified, :type => "boolean ()"
     publish :function => :SetModified, :type => "void (boolean)"
     publish :function => :Save, :type => "void ()"
+    publish :function => :reset, :type => "void ()"
```
Comment (joseivanlopez) [inline keyboard/src/modules/Keyboard.rb]: Do we still need to publish methods?

File: keyboard/src/modules/Keyboard.rb [hunk 2]
```diff
@@ -116,6 +116,15 @@ def SetModified
       @modified = true
     end
 
+    # Reset all setting
+    def reset
```
Comment (joseivanlopez) [inline keyboard/src/modules/Keyboard.rb]: Where is this used? Please, document why it is needed.

Comment (schubi2) [reply keyboard/src/modules/Keyboard.rb]: For the testcase only. Or do you see another chance to reset the current keyboard ?

Comment (joseivanlopez) [reply keyboard/src/modules/Keyboard.rb]: What about defining private attr_readers and then mock it?

~~~
private

attr_reader :curr_kbd
attr_reader :keyboard_on_entry
attr_reader :default_kbd
attr_reader :user_decision
attr_reader :modified

# In the test

allow(subject).to receive(:curr_kbd).and_return("")
~~~

Comment (joseivanlopez) [reply keyboard/src/modules/Keyboard.rb]: And make sure to use `!curr_kbd.empty?` instead of `!@curr_kbd.empty?` in the module code.

**Comment by joseivanlopez** [comment]
> Please, try this https://github.com/yast/yast-country/pull/231/files#r352557334

Finally we decided to simply use a new instance in the test:

~~~
subject { Yast::KeyboardClass.new }
~~~

**Pull request merged**